Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,350 @@
+
+  return rotPsi * rotTheta * newDirectionHep;
+}
+
+template <typename generator_t>
+std::vector<Particle> PhotonConversion::recordProduct(
+    generator_t& generator, const Particle& photon, Scalar childEnergy,
+    const Particle::Vector3& childDirection) const {
+  // Calculate the child momentum
+  const Scalar massChild = findMass(Acts::eElectron);
+  const Scalar momentum1 =
+      sqrt(childEnergy * childEnergy - massChild * massChild);
+
+  // Use energy-momentum conservation for the other child
+  const Particle::Vector3 vtmp =
+      photon.fourMomentum().template segment<3>(Acts::eMom0) -
+      momentum1 * childDirection;
+  const Scalar momentum2 = vtmp.norm();
+
+  // Charge sampling
+  std::uniform_int_distribution<> uniformDistribution{0, 1};
+  Scalar charge1;
+  Scalar charge2;
+  if (uniformDistribution(generator) == 1) {
+    charge1 = -1.;
+    charge2 = 1.;
+  } else {
+    charge1 = 1.;
+    charge2 = -1.;
+  }
+
+  // Assign the PDG ID
+  const Acts::PdgParticle pdg1 =
+      static_cast<Acts::PdgParticle>(std::copysign(Acts::eElectron, charge1));
+  const Acts::PdgParticle pdg2 =
+      static_cast<Acts::PdgParticle>(std::copysign(Acts::eElectron, charge2));
+
+  // Build the particles and store them
+  std::vector<Particle> children;
+  children.reserve(2);
+
+  if (momentum1 > childMomentumMin) {
+    Particle child = Particle(Barcode(), pdg1)
+                         .setPosition4(photon.fourPosition())
+                         .setDirection(childDirection)
+                         .setAbsoluteMomentum(momentum1)
+                         .setProcess(ProcessType::ePhotonConversion);
+    children.push_back(std::move(child));
+  }
+  if (momentum2 > childMomentumMin) {
+    Particle child = Particle(Barcode(), pdg2)
+                         .setPosition4(photon.fourPosition())
+                         .setDirection(childDirection)
+                         .setAbsoluteMomentum(momentum2)
+                         .setProcess(ProcessType::ePhotonConversion);
+    children.push_back(std::move(child));
+  }
+  return children;
+}
+
+template <typename generator_t>
+bool PhotonConversion::run(generator_t& generator, Particle& particle,
+                           std::vector<Particle>& generated) const {
+  // Fast exit if particle is not alive
+  if (!particle)
+    return true;
+
+  // Fast exit if momentum is too low
+  const Scalar p = particle.absoluteMomentum();
+  if (p < 2. * childMomentumMin)
+    return false;
+
+  // Get one child energy
+  const Scalar childEnergy = p * childEnergyFraction(generator, p);
+
+  // Now get the deflection
+  const Particle::Vector3 childDir =
+      childDirection(generator, particle.fourMomentum());
+
+  // Produce the final state
+  const std::vector<Particle> finalState =
+      recordProduct(generator, particle, childEnergy, childDir);
+  generated.insert(generated.end(), finalState.begin(), finalState.end());
+
+  // Kill the photon
+  particle.setAbsoluteMomentum(0.);
```

Comment:
This is a bit tricky to explain. You can kill the particle in the `Interactor`, but although you end the propagation, the `Propagator` calls once more the action list. If you just declare a particle as not alive then the zombie-particles will perform an additional conversion since all conditions are fulfilled. By explicitly killing the particles, I can prevent it.

Reply:
#691 fixes this issue - it stays alive in here